components/movementsensor/adxl345/interrupts.go

```diff
@@ -0,0 +1,80 @@
+package adxl345
+
+// addresses relevant to interrupts.
+const (
+	// contains an unsigned time value representing the maximum time that an event must be above the
+	// THRESH_TAP threshold to qualify as a tap event [625 µs/LSB].
+	DurAddr byte = 0x21
+	// contains info on which interrupts have been enabled.
+	IntEnableAddr byte = 0x2E
+	// contains info on which interrupt pin to send each interrupt.
+	IntMapAddr byte = 0x2F
+	// contains info on which interrupt has gone off since the last time this address has been read from.
+	IntSourceAddr byte = 0x30
+	// contains an unsigned time value representing the wait time from the detection of a tap event to the
+	// start of the time window [1.25 ms/LSB].
+	LatentAddr byte = 0x22
+	// contains info on which axes have been turned on for taps (X, Y, Z are bits 2, 1, 0 respectivel).
+	TapAxesAddr byte = 0x2A
+	// contains an unsigned threshold value for tap interrupts [62.5 mg/LSB ].
+	ThreshTapAddr byte = 0x1D
+	// contains the threshold value, in unsigned format, for free-fall detection.
+	ThreshFfAddr byte = 0x28
+	// containsthe  minimum time that the value of all axes must be less than THRESH_FF to generate a free-fall interrupt.
+	TimeFfAddr byte = 0x29
+)
+
+// types of interrupts.
+const (
+	SingleTap string = "SINGLE_TAP"
+	FreeFall  string = "FREE_FALL"
```

Comment:
I'm going to suggest something I'm not sure I believe in. Is this worth changing? Maybe not.

Making these strings will make them easily readable by a human if you print them out, but will slow down all computations (e.g., putting them in a map means that looking them up again requires comparing 9-10 bytes, when comparing integers could be done much more quickly.

```go
type InterruptID = uint8
const (
    SingleTap InterruptID = iota
    FreeFall InterruptID = iota
)

var interruptBitPosition = map[InterruptID]byte {
    SingleTap: 1 << 6,
    FreeFall: 1 << 2,
}

// If you ever want to print out the names, use this.
var interruptNames = map[InterruptID]string {
    SingleTap: "SINGLE_TAP",
    FreeFall: "FREE_FALL",
}
...
```

This will definitely at least as fast. Will it actually be faster? I'm not sure. Will the difference be important? In this specific example, I doubt it.

Reply:
Happy to change this if people think it is a good idea! What is the purpose of setting them to `iota`?

Reply:
By making them numbers instead of strings, comparisons take 1 clock cycle instead of a dozen, and I suspect that looking something up in a map goes significantly faster, too. but this specific example doesn't need to be very performant, and I doubt anyone will notice if we leave them as strings. It's more just a good habit to get into, for the times when the code will run a whole lot.